src/main/java/org/shanerx/tradeshop/commands/CommandCaller.java

```diff
@@ -154,4 +160,18 @@ public boolean checkPerm() {
 
         return true;
     }
+
+    /**
+     * Checks if the sender has the required permission
+     *
+     * @return true if permission is NONE or sender has permission
+     */
+    public boolean checkPerm(TradeCommand command) {
+        if (!cmdPass.getSender().hasPermission(command.getPermission()) && !command.getPermission().equals(Permissions.NONE)) {
```

Comment:
Can we make this check if the add-on adds a permission for the command as well?

Reply:
What do you mean? command.getPermission() is the permission for the subcommand, provided by the addon.

Reply:
but this method is checking against the command enum is it not? If it is, it isnt actually checking for the add-on sub commands

Reply:
I don’t understand what you mean. The function TradeCommand#getPermission() is most certainly meant to return the permission for that command.

Reply:
I see now, didnt register that its using TradeCommand not Commands. can we change the variable to tradeCommand instead of command so that its a little easier to read?

Reply:
Yes, now there are two `checkPerms` methods, distibguished by their argument types.

Reply:
yeh, i think changing the variable names will make it easier to tell the difference between the 2 methods when glancing at them

Reply:
Ok, will do.